### ISSUE TITLE
Transpile for non-modern browsers

### ISSUE THREAD

**Issue by isBatak** [bartlomiejzuber/use-open-window#1]
Could you use https://github.com/developit/microbundle to support older browsers?
It seems this lib is not working on browsers that don't support spread operators, like old Edge and IE11.

**Comment by bartlomiejzuber** [comment]
Hey, thanks for letting me know. I will check this one out. 👍

**Comment by bartlomiejzuber** [comment]
@isBatak Fixed in `1.2.1` version. You can update it via `yarn upgrade use-open-window@1.2.1`.
I've checked the lib you've proposed but according to latest issue (https://github.com/developit/microbundle/issues/565) this lib have really ugly bug inside and while the change you require can be covered by simple change in tsconfig I've choose that way.

Thank you for you're contribution here.

**Comment by isBatak** [comment]
@bartlomiejzuber maybe you can check this one https://github.com/jaredpalmer/tsdx. It's comparable with microbundle https://github.com/jaredpalmer/tsdx/blob/master/README.md#comparison-to-microbundle. 
At least you should setup Rollup bundler to support all module types (and there are few of them https://dev.to/iggredible/what-the-heck-are-cjs-amd-umd-and-esm-ikm).

I'm using your lib with Next.js so it should support cjs (node), umd(old browsers) and esm (modern). https://github.com/zeit/next.js/issues/7563
I'm not sure tsc can handle them all.

**Comment by isBatak** [comment]
This should work also.
```
"main": "./dist/index.js",
"module": "./esm/index.js",
"types": "./dist/index.d.ts",
"files": [
    "dist/**",
    "esm/**"
],
"scripts": {
    "build": "npm run build:esm && npm run build:cjs",
    "build:cjs": "ncc build src/index.ts -o dist -m -e react",
    "build:esm": "tsc --target ESNext --module ES6 --outDir esm",
},
...
```
ncc is this lib https://www.npmjs.com/package/@zeit/ncc

**Comment by bartlomiejzuber** [comment]
Sounds great, I've used lib https://github.com/jaredpalmer/tsdx. All mentioned module types has been shipped inside `1.2.2` version. Here's how /dist looks like.
![image](https://user-images.githubusercontent.com/9931428/76902443-4d57ad80-689c-11ea-9f03-eb44391901a5.png)

**Comment by isBatak** [comment]
Wooohoo!!! Thx! You are the best! :D

**Comment by isBatak** [comment]
@bartlomiejzuber I think you need to add these three settings to `package.json` also
```
"module": "dist/use-open-window.esm.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist"
  ],
```

**Comment by bartlomiejzuber** [comment]
@isBatak It's done and published under `1.2.3` version. :) Glad that I could help and I learnt something as well. Thanks!
